website/project/model.py

```diff
@@ -2359,15 +2371,19 @@ def add_log(self, action, params, auth, foreign_user=None, log_date=None, save=T
             user=user,
             foreign_user=foreign_user,
             params=params,
+            node=self,
+            original_node=params['node']
         )
 
         if log_date:
             log.date = log_date
+        log.save()
 
-        self.date_modified = log.date.replace(tzinfo=None)
+        if len(self.logs) == 1:
```

Comment:
This conditional appears to be for optimization, but the `.logs` property already evaluates the MODM queryset to return a list, so I'm not sure that this saves any time. Just 
`self.date_modified = self.logs[-1].date.replace(tzinfo=None)` might be better.